### PR TITLE
add `customProvider` property for better native ethers support

### PR DESCRIPTION
**Pull request by NoahZinsmeister**

### CODE REVIEW

File: packages/core/src/hooks.ts [hunk 1]
```diff
@@ -320,7 +320,11 @@ function getAugmentedHooks<T extends Connector>(
   { useChainId, useAccounts }: ReturnType<typeof getStateHooks>,
   { useAccount, useIsActive }: ReturnType<typeof getDerivedHooks>
 ) {
-  function useProvider(network?: Networkish, enabled = true): Web3Provider | undefined {
+  // the default type is set to Web3Provider so that consumers who don't use customProvider aren't impacted
```
Comment (zzmp) [inline packages/core/src/hooks.ts:323]: I'd rephrase this as `// Avoid type erasure by returning the most qualified type if not otherwise set`

File: packages/network/src/index.ts [hunk 2]
```diff
@@ -1,21 +1,19 @@
...
 type url = string | ConnectionInfo
 
-function parseChainId(chainId: string) {
-  return Number.parseInt(chainId, 16)
-}
-
 export class Network extends Connector {
   /** {@inheritdoc Connector.provider} */
-  public provider: Eip1193Bridge | undefined
+  public provider: undefined
```
Comment (zzmp) [inline packages/network/src/index.ts:10]: Does `.provider` need to be specified at all given that `.customProvider` will be?

File: packages/network/src/index.ts [hunk 3]
```diff
@@ -73,15 +67,9 @@ export class Network extends Connector {
   public async activate(desiredChainId = this.defaultChainId): Promise<void> {
     if (!this.provider) this.actions.startActivation()
 
-    this.provider = await this.isomorphicInitialize(desiredChainId)
+    this.customProvider = await this.isomorphicInitialize(desiredChainId)
```
Comment (zzmp) [inline packages/network/src/index.ts:70]: Can this promise reject? If so, `.catch` and `this.actions.reportError`.

File: packages/types/src/index.ts [hunk 4]
```diff
@@ -77,6 +77,12 @@ export abstract class Connector {
    */
```
Comment (zzmp) [inline packages/types/src/index.ts:77]: The above doc comment should be amended, as "This property must be defined while the connector is active." is no longer true.

File: packages/url/src/index.ts [hunk 5]
```diff
@@ -1,17 +1,15 @@
...
 type url = string | ConnectionInfo
 
-function parseChainId(chainId: string) {
-  return Number.parseInt(chainId, 16)
-}
-
 export class Url extends Connector {
   /** {@inheritdoc Connector.provider} */
-  public provider: Eip1193Bridge | undefined
+  public provider: undefined
```
Comment (zzmp) [inline packages/url/src/index.ts:10]: Does `.provider` need to be specified at all given that `.customProvider` will be?

File: packages/url/src/index.ts [hunk 6]
```diff
@@ -51,12 +45,7 @@ export class Url extends Connector {
     await this.isomorphicInitialize()
```
Comment (zzmp) [inline packages/url/src/index.ts:45]: Can this promise reject? If so, `.catch` and `this.actions.reportError`.

File: packages/url/src/index.ts [hunk 7]
```diff
@@ -35,12 +33,8 @@ export class Url extends Connector {
   private async isomorphicInitialize() {
```
Comment (zzmp) [inline packages/url/src/index.ts:33]: network returned the provider to be set as part of activate, and it felt cleaner (mostly because error propagation could be handled in the activate method).

File: packages/network/src/index.ts (hunk 2 above)
Comment (NoahZinsmeister) [reply packages/network/src/index.ts:10]: ah, i'm just narrowing the type of the parent class so that it can never be set

File: packages/url/src/index.ts (hunk 5 above)
Comment (NoahZinsmeister) [reply packages/url/src/index.ts:10]: ditto

File: packages/network/src/index.ts (hunk 3 above)
Comment (NoahZinsmeister) [reply packages/network/src/index.ts:70]: neither this nor the other instance should be able to throw, but on the off chance that ethers errors in the constructor of JsonRpcProvider or something, i'll add a catch!
